crates/matrix-sdk/src/widget_api/capabilities.rs

```diff
@@ -28,20 +32,30 @@ pub struct ReadEventRequest {
     pub description: EventDescription,
 }
 
+impl From<RawReadEventRequest> for ReadEventRequest {
+    fn from(req: RawReadEventRequest) -> Self {
+        Self {
+            limit: req.limit,
+            description: EventDescription {
+                event_type: req.message_type,
+                kind: if req.state_key.is_empty() {
+                    EventKind::Timeline
+                } else {
+                    EventKind::State { key: req.state_key }
+                },
+            },
+        }
+    }
+}
```

Comment:
The `From` defines a conversion from `T` to `W`, where `T` is a raw event and `W` is a strongly typed version of it. We use it in one place at the moment if I'm not mistaken (the handler works with higher-level types rather than with low-level primitives). Though, ideally, it's probably a not bad idea if we deserialized it in a more strictly typed way to begin with. I hope I could address the question :)

Reply:
I see, yeah, maybe you're right and this sort of type is not required.

Reply:
Done.